tests/test_stdin_encoding_1426.py

```
@@ -0,0 +1,713 @@
+    assert read_payload_text(_Bytes(b""), None) == ""
+    assert read_payload_text(_Bytes(PAYLOAD_UTF8), None) == PAYLOAD_TEXT
+
+
+# --- The archive password: the irreversible half ------------------------
+
+# Deliberately chosen so cp1252 can decode it *without raising*. Every
+# byte of "café".encode("utf-8") (63 61 66 c3 a9) is assigned in cp1252,
+# so the wrong decode succeeds and yields "cafÃ©" — a different password,
+# silently. A test that only asserted "no exception" would pass on the
+# unfixed tree.
+PASSWORD = "café-passphrase"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "PASSWORD"

(S105)